src/Agent.Listener/JobDispatcher.cs

```diff
@@ -602,6 +602,7 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, WorkerDisp
                                 var messageType = MessageType.CancelRequest;
                                 if (HostContext.AgentShutdownToken.IsCancellationRequested)
                                 {
+                                    resultOnAbandonOrCancel = TaskResult.Failed;
```

Comment:
Yes, if the user terminates the Agent using a SIGTERM or SIGINT signal, it will mark the Job as Failed. I did test it and if the User cancels the Pipeline from ADO, the job will be marked as Canceled (no change in behavior)